app/views/users/show.html.erb

```diff
@@ -89,17 +89,32 @@
 
         <md-input-container>
           <label>Full Name</label>
-          <input ng-model="user.name" ng-change="user.changed = true">
+          <input name="fullname" required ng-model="user.name" md-maxlength="15" ng-change="user.changed = true">
+          <div ng-messages="fullname.$error"></div>
+          <div ng-messages="!fullname.$error">
+            <div ng-message="required">This is required.</div>
+            <div ng-message="md-maxlength">The user name must be less than 35 characters long.</div>
+        </div>
         </md-input-container>
 
         <md-input-container>
           <label>Email Address</label>
-          <input ng-model="user.params.email.value" ng-change="user.changed = true">
+          <input name='email' ng-model="user.params.email.value" ng-change="user.changed = true" minlength="10" maxlength="100" ng-pattern="/^.+@.+\..+$/">
```

Comment:
I think it is possible to have an email address that is shorter than 10 characters. My bet is 6 is the lower bound.